src/app/features/date-range/components/date-range/date-range.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, ChangeDetectionStrategy, Input, Inject } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { DateRange } from '@angular/material/datepicker';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { DateRangeData, UnavailableDates } from '../../models';
+import { DateRangeController } from '../../services';
+
+@Component({
+  selector: 'lc-date-range',
+  templateUrl: './date-range.component.html',
+  styleUrls: ['./date-range.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateRangeComponent {
+  @Input() selectedRangeValue!: DateRange<Date> | null;
+
+  readonly maxRentalPeriod = 10;
+  readonly day = new Date().getDate();
+  readonly month = new Date().getMonth();
+  readonly year = new Date().getFullYear();
+
+  dateRangeForm = new FormGroup({
+    fromDate: new FormControl(null),
+    toDate: new FormControl(null),
+  });
+
+  minDate = new Date();
+  maxDate = new Date(this.year + 1, this.month, this.day - 1);
+
+  private unavailableDates: UnavailableDates[] = this.dateRangeData.unavailableDates;
+
+  constructor(@Inject(MAT_DIALOG_DATA) public dateRangeData: DateRangeData, private controller: DateRangeController) {}
+
+  isDateUnavailable = (dateFromCalendar: Date): boolean => {
+    const timeUnixFromCalendar = dateFromCalendar.setHours(0, 0, 0, 0);
```

Comment:
since date it is an object you can mutate it here accidentally. and then it can affect some how. I would create a function like removeTime(data: Date): Date which takes a date object, creates new dates object, sets hours for new objects and return it. Use it everywhere  where you need to remove hours

Reply:
done